File: backend/src/main/java/com/inn/cafe/Service/UserService.java
package com.inn.cafe.Service;

import com.inn.cafe.Wrapper.UserWrapper;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Map;

public interface UserService {

    ResponseEntity<String> signUp(Map<String, String> requestMap);

    ResponseEntity<String> login(Map<String, String> requestMap);

    ResponseEntity<List<UserWrapper>> getAllUser();  // to get all the user by role (admin or user)

    ResponseEntity<String> update(Map<String, String> requestMap); // to update the information of users

    ResponseEntity<String> checkToken();   // to check the token

    ResponseEntity<String> changePassword(Map<String, String> requestMap);  // to change the password


    // method for FORGET PASSWORD using @RequestBody 👇👇👇👇👇
    ResponseEntity<String> forgotPassword(Map<String, String> requestMap);


//    // OR method for FORGET PASSWORD using @RequestParam 👇👇👇👇👇
//    ResponseEntity<String> forgotPassword(String email);

}

File: frontend/src/services/userService.ts
import api from './api';

export interface IUser {
  id: number;
  name: string;
  email: string;
  contactNumber: string;
  status: string;
  role: string;
}

export interface ILoginRequest {
  email: string;
  password: string;
}

export interface ISignupRequest {
  name: string;
  contactNumber: string;
  email: string;
  password: string;
}

export interface IChangePasswordRequest {
  oldPassword: string;
  newPassword: string;
}

const userService = {
  login: (data: ILoginRequest) => {
    return api.post<string>('/user/login', data);
  },

  signup: (data: ISignupRequest) => {
    return api.post<string>('/user/signup', data);
  },

  getAllUsers: () => {
    return api.get<IUser[]>('/user/get');
  },

  updateUserStatus: (userId: number, status: string) => {
    return api.post<string>('/user/update', { id: userId, status });
  },

  checkToken: () => {
    return api.get<string>('/user/checkToken');
  },

  changePassword: (data: IChangePasswordRequest) => {
    return api.post<string>('/user/changePassword', data);
  },

  forgotPassword: (email: string) => {
    return api.post<string>('/user/forgotPassword', { email });
  }
};

export default userService;
